code/xcvm/SPEC.md

```diff
@@ -249,9 +254,9 @@ Call {
 
 On the executing interpreter, `BindingValue::Self` will be interpolated at byte index 13  of the payload before being executed, the final payload then becomes `swap(10,(1,2), BindingValue::Self)`, where `BindingValue::Self` is the canonical address of the interpreter on the destination side.
 
-Besides accessing the `Self` register, `BindingValue` allows for lazy lookups of AssetId conversions, by using `BindingValue::AssetId(GlobalId)`, or lazily converting decimal points depending on the chain using the `Balance` type.
+Besides accessing the `Self` register, `BindingValue` allows for lazy lookups of AssetId conversions, by using `BindingValue::AssetId(GlobalId)`, or lazily converting amount value depending on the chain using the `AssetAmount` type.
 
-Bindings do not support non byte aligned encodings.
+Bindings do not support non byte aligned encodings. To correctly bind values in the execution layer, the binding values should be ordered by their position index(u16) respectively since to process them interpreters will iterate over the bindings in sequence, after each iteration, the original payload will be updated with the new binding value and the next binding value position will be shifted by the length of the binding value. 
```

Comment:
```suggestion
Bindings do not support non-byte-aligned encodings. To correctly bind values in the execution layer, the binding values should be ordered by their position index(u16) respectively. Because to process them, interpreters will iterate over the bindings in sequence. After each iteration, the original payload will be updated with the new `BindingValue` and the next `BindingValue` position will be shifted by the length of the `BindingValue`. 
```